### PR TITLE
Replace cilium network policy and update proxy tests

### PR DESCRIPTION
**Pull request by chrismuellner**
# Description

Update cilium network policies to default network policies:
* Update documentation to no longer require cilium (replaced with istio)
* Remove todos for cilium

Update checks in istio/proxy/network tests to work:
* Remove curl pod manifest and create curl pod in code (most of the content was replaced anyway)
* Update proxy usage to `https_proxy` environment variable
* Update proxy availability check to reflect changes
  * webhook should show `CONNECT <webhook service>:443`
  * activegate should show `RUNNING`
* Update proxy annotations to disable injection to combined annotation

## How can this be tested?
Following e2e tests, that were previously skipped, should pass:
* `make tests/e2e/cloudnative/istio`
* `make tests/e2e/cloudnative/network`
* `make tests/e2e/cloudnative/proxy`
* `make tests/e2e/activegate/proxy`

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: test/scenarios/README.md [hunk 1]
```diff
@@ -116,7 +116,7 @@ installed and pre-configured on the cluster.
 
 # CloudNative - network
 ## Prerequisites
-cilium service mesh
+istio service mesh
```
Comment (luhi-DT) [inline test/scenarios/README.md:119]: Should we mention in the Readme how to install istio locally with helm?

**Review by luhi-DT — APPROVED** [review]

File: test/scenarios/README.md (hunk 1 above)
Comment (chrismuellner) [reply test/scenarios/README.md:119]: we could link to the official documentation: https://istio.io/latest/docs/setup/install/helm/#installation-steps

Comment (luhi-DT) [reply test/scenarios/README.md:119]: fine for me, just a hint what to install and what is needed is fine

Comment (chrismuellner) [reply test/scenarios/README.md:119]: documentation will be improved in a follow up

**Pull request merged**